net/zerotier/files/lib/netifd/proto/zerotier.sh

```diff
@@ -0,0 +1,173 @@
+#!/bin/sh
+
+DAEMON=/usr/bin/zerotier-one
+CONFIG_PATH=/var/lib/zerotier-one
+
+[ -n "$INCLUDE_ONLY" ] || {
+	. /lib/functions.sh
+	. ../netifd-proto.sh
+	init_proto "$@"
+}
+
+proto_zerotier_init_config() {
+	available=1
+	no_device=1
+	proto_config_add_string "network_id"
+	proto_config_add_string "secret"
+	proto_config_add_string "port"
+	proto_config_add_string "config_path"
+	proto_config_add_boolean "copy_config_path"
+}
+
+proto_zerotier_setup() {
+	local interface="$1"
+
+	local network_id secret port config_path copy_config_path
+	local args=""
+
+	config_load network
+	config_get network_id "${interface}" "network_id"
+	config_get secret "${interface}" "secret"
+	config_get port "${interface}" "port"
+	config_get config_path "${interface}" "config_path"
+	config_get_bool copy_config_path "${interface}" "copy_config_path" 0
+
+	path=${CONFIG_PATH}_$interface
+
+	# Remove existing link or folder
+	rm -rf "${path}"
+
+	# Create link or copy files from CONFIG_PATH to config_path
+	if [ -n "$config_path" -a "$config_path" != "$path" ]; then
+		if [ ! -d "$config_path" ]; then
+			echo "ZeroTier config_path does not exist: $config_path" 1>&2
+			return
+		fi
+
+		# ensure that the target exists
+		mkdir -p $(dirname "${path}")
+
+		if [ "$copy_config_path" = "1" ]; then
+			cp -r "$config_path" "${path}"
+		else
+			ln -s "$config_path" "${path}"
+		fi
+	fi
+
+	mkdir -p "${path}"/networks.d
+	touch "${path}"/networks.d/"${network_id}".conf
+
+	# enforce service to use given interface name
+	touch "${path}"/devicemap
+	echo "${network_id}=${interface}" >> "${path}"/devicemap
+
+	if [ -n "$port" ]; then
+		args="$args -p${port}"
+	else
+		args="$args -p0"
+	fi
+
+	local secret
+	secret="$(uci get network."${interface}".secret)"
+
+	if [ -z "$secret" ]; then
+		echo "Generate secret - please wait..."
+		local sf="/tmp/zt.${interface}.secret"
+
+		zerotier-idtool generate "$sf" > /dev/null
+		[ $? -ne 0 ] && return 1
+
+		secret="$(cat $sf)"
+		rm "$sf"
+
+		uci set network."${interface}".secret="$secret"
+		uci commit network
+	fi
+
+	if [ -n "$secret" ]; then
+		echo "$secret" > "${path}"/identity.secret
+		# make sure there is not previous identity.public
+		rm -f "${path}"/identity.public
+	fi
+
+	proto_run_command "${interface}" "${DAEMON}" -p0 "${path}"
+
+	sleep 1
+
+	local status
+	while true; do
+		status="$(zerotier-cli -D${path} get ${network_id} status)" || ""
```

Comment:
Not sure of `|| ""` is needed. But maybe a redirect of stderr to /dev/null might be a good idea.